### PR TITLE
CORDA-966 - RELEASE - Merge release notes from release branch to master

### PR DESCRIPTION
**Pull request by fenryka**

### CODE REVIEW

File: docs/source/release-notes.rst [hunk 1]
```diff
@@ -1,49 +1,301 @@
...
+
+  .. important:: This replaces the Network Map service that was present in Corda 1.0 and Corda 2.0.
+
+  Further information can be found in the :doc:`changelog`, :doc:`network-map` and :doc:`setting-up-a-corda-network` documentation.
+
+* **Contract Upgrade**
+
+  Support for the upgrading of contracts has been significantly extended in this release.
+
+  Contract states express which attached JARs can define and check them using _constraints_. In older versions the only supported
```
Comment (josecoll) [inline docs/source/release-notes.rst]: and **verify** (not check)

File: docs/source/release-notes.rst [hunk 2]
```diff
@@ -1,49 +1,301 @@
...
+    Corda 3.0 lays the groundwork for future releases, when contract verification will be done against the attached contract JARs
+    rather than requiring a locally deployed CorDapp of the exact version specified by the transaction. The future vision for this
+    feature will entail the dynamic downloading of the appropriate version of the smart contract and its execution within a
+    sandboxed environment.
+
+    .. warning:: This change means that your app JAR must now fit inside the 10mb attachment size limit. To avoid redundantly copying
+      unneeded code over the network and to simplify upgrades, consider splitting your application into two or more JARs - one that
+      contains states and contracts (which we call the app "kernel"), and another that contains flows, services, web apps etc. For
+      example, our `Cordapp template <https://github.com/corda/cordapp-template-kotlin/tree/release-V3>`_ is structured like that.
+      Only the first will be attached. Also be aware that any dependencies your app kernel has must be bundled / "sharded" into a fat JAR,
```
Comment (josecoll) [inline docs/source/release-notes.rst]: please remove the work "sharded" (bundled is just fine)

File: docs/source/release-notes.rst [hunk 3]
```diff
@@ -1,49 +1,301 @@
...
+    If this step fails the normal transaction verification failure path is followed.
+
+    Corda 3.0 lays the groundwork for future releases, when contract verification will be done against the attached contract JARs
+    rather than requiring a locally deployed CorDapp of the exact version specified by the transaction. The future vision for this
+    feature will entail the dynamic downloading of the appropriate version of the smart contract and its execution within a
+    sandboxed environment.
+
+    .. warning:: This change means that your app JAR must now fit inside the 10mb attachment size limit. To avoid redundantly copying
+      unneeded code over the network and to simplify upgrades, consider splitting your application into two or more JARs - one that
+      contains states and contracts (which we call the app "kernel"), and another that contains flows, services, web apps etc. For
```
Comment (josecoll) [inline docs/source/release-notes.rst]: Should we really be introducing new terms here: "app kernel".
This new term should be clearly defined (under terminology) and used consistently in the general Corda documentation.

File: docs/source/release-notes.rst [hunk 4]
```diff
@@ -1,49 +1,301 @@
 Release notes
 =============
 
-Here are release notes for each snapshot release from M9 onwards.
+Release 3.0
+-----------
+
+Corda 3.0 is here and brings with it wire stability and a host of other exciting features that aim to enhance
```
Comment (josecoll) [inline docs/source/release-notes.rst]: "is here" adds no significance (and will be outdated as soon as the next release is made).

File: docs/source/release-notes.rst [hunk 5]
```diff
@@ -1,49 +1,301 @@
...
+the developer and user experience whilst providing for the long term usability of deployed Corda instances. It
+will provide functionality to ensure anyone wishing to move to the anticipated release of R3 Corda can do so
```
Comment (josecoll) [inline docs/source/release-notes.rst]: I would not made any reference to a specific "anticipated release of R3 Corda".
It would be better to simply state that this release will enable mixed networks of "interoperable and upgradable Corda and R3 Corda nodes".

File: docs/source/release-notes.rst (hunk 4 above)
Comment (josecoll) [inline docs/source/release-notes.rst]: I would also drop the "a host of other exciting features" (too wordy and unspecific).
The 1st paragraph should emphasize "interoperability and upgradability" (wire, data, contract and node upgrades) as a foundation for future mixed Corda networks.

Comment (fenryka) [reply docs/source/release-notes.rst]: We said the same for version 1 though, it's... designed to be read in the moment

File: docs/source/release-notes.rst (hunk 5 above)
Comment (fenryka) [reply docs/source/release-notes.rst]: RGB added that so I'm probably going to keep it, we want to drive some hype for R3C3

File: docs/source/release-notes.rst (hunk 1 above)
Comment (fenryka) [reply docs/source/release-notes.rst]: done

File: docs/source/release-notes.rst (hunk 2 above)
Comment (fenryka) [reply docs/source/release-notes.rst]: done

File: docs/source/release-notes.rst (hunk 3 above)
Comment (fenryka) [reply docs/source/release-notes.rst]: Its Mike H's  but removing the brackets won't hurt
